staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+)
+
+var correctConfig string = `
+- kind: k8s-aes-gcm
+  version: v1
+  keys:
+    - name: key1
```

Comment:
If we keep an `unencrypted-key`, we may have to add a special case to not add the `<provider-name>:` prefix to the data while doing TransformToStorage for cases when data is unencrypted. Rather, it may be preferable to use a separate provider to achieve that.

Something like this (if I've understood this correctly):
```
- kind: identity
  version: v1
  resource: namespaces,secrets
- kind: k8s-aes-gcm
  version: v1
  keys:
    - name: key1
      secret: c2VjcmV0IGlzIHNlY3VyZQ==
    - name: key2
      secret: dGhpcyBpcyBwYXNzd29yZA==
  resource: namespaces,secrets,serviceaccounts
```

This way, identity transformer would be used for namespaces and secrets while encrypting to disk. But while reading from disk, we can still use k8s-aes, which seems to be the desired option. We can later add an endpoint to re-store all the data with the updated transformers, which would do a complete migration. That would be useful for rotation as well.